packages/devtools_app/lib/src/timeline/timeline_flame_chart.dart

```diff
@@ -20,6 +21,97 @@ import '../utils.dart';
 import 'timeline_controller.dart';
 import 'timeline_model.dart';
 
+class TimelineFlameChartContainer extends StatefulWidget {
+  const TimelineFlameChartContainer({
+    @required this.processing,
+    @required this.processingProgress,
+  });
+
+  @visibleForTesting
+  static const emptyTimelineKey = Key('Empty Timeline');
+
+  final bool processing;
+
+  final double processingProgress;
+
+  @override
+  _TimelineFlameChartContainerState createState() =>
+      _TimelineFlameChartContainerState();
+}
+
+class _TimelineFlameChartContainerState
+    extends State<TimelineFlameChartContainer>
+    with AutoDisposeMixin {
+  TimelineController controller;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    final newController = Provider.of<TimelineController>(context);
+    if (newController == controller) return;
+    controller = newController;
+  }
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
The code in this build method was moved from timeline_screen.dart. The areaPaneHeader addition and rearranging of the OutlineDecoration is the only change that is not copy paste.